src/sqlfluff/cli/commands.py

```diff
@@ -247,7 +247,7 @@ def core_options(f: Callable) -> Callable:
     )(f)
     f = click.option(
         "--encoding",
-        default="autodetect",
+        default=None,
```

Comment:
Providing an explicit default value of `"autodetect"` was causing any setting in `.sqlfluff` to be overridden by the command-line value. (You'll note that elsewhere in this file, command-line options that override `.sqlfluff` values already had a default value of `None`.)